server/source/serverlib/game/game.cpp

```diff
@@ -105,6 +115,45 @@ void Game::doCurrentPlayerBuyField()
     m_board.changeOwner(propertyId, currentPlayer().name());
 }
 
+void Game::doCurrentPlayerBuyHouse(int fieldID, bool buy)
+{
+    auto buyPrice = m_board.housePrice(fieldID);
+    auto sellPrice = buyPrice / 2;
+
+    m_board.ensureFullGroupOwnership(currentPlayer().name(), fieldID);
+
+    if (buy)
+    {
+        if (m_bank.balance(currentPlayer().name()) < buyPrice)
```

Comment:
Can be removed if you first take the money. Then build the house.

Reply:
The problem is that both operations can fail, and we want neither to complete if any of them fails. Without additional checks, if the second operation (whatever it may be) fails, the first operation will already be completed.